tools/scripts/release/parseCrates.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Example Usage:
+# cargo deny --all-features list --config=tools/cargo-deny/deny.toml --layout=crate --format json | ./tools/scripts/release/parseCrates.sh > NOTICE
+
+OPEN_BRACKET=$(expr 0)      # count open brackets, diff between a license name and library name
+                            # set to 1 as we remove the 1st instance before counting
+LICENSE_STRING=""
+FILEOUT=""
+PROJECT=""
+VERSION=""
+URL=""
+
+OLD_CKSUM=$(cat NOTICE | cksum)         # cksum old file, compare value to new below
+
+FILEOUT=$(printf "This file contains attributions for any 3rd-party open source code used in this project.\n")
+FILEOUT+=$(printf "\n\n\n\n%1s %-32s %1s %-55s %1s %-80s %1s" "|" "Name " "|" "License " "|" "URL" "|" )
+FILEOUT+=$(printf "\n|----------------------------------|---------------------------------------------------------|----------------------------------------------------------------------------------|")
+
+# Derive input
+INPUT=$(jq "." $1 | tr -d '{},()":')
+
+# Set IFS to newline
+IFS=$'\n'
+
+for line in $INPUT
+do
+
+    # We have the closing bracket, set OPEN_BRACKET to 0, remove any leading commas from
+    # licensing, and build output for this library & licensing
+    # NOTE: Important this check is before check for '[', otherwise crates w/o license
+    # are presented incorrectly
+    if [[ $line == *"]"* ]]; then
+        IFS=$' '
+        OPEN_BRACKET=$(expr 0)      # reset OPEN_BRACKET status
+        LICENSE_STRING="${LICENSE_STRING#,}"    # remove any leading comma
+        LICENSE_STRING=$(echo $LICENSE_STRING | sed 's/ //g')       # remove leading spaces from LICENSES
+        LICENSE_STRING=$(echo $LICENSE_STRING | sed 's/,/, /g')     # add space after commas in LICENSES
+        FILEOUT+=$(printf "\n%1s %-32s %1s %-55s %1s %-80s %1s" "|" "$PROJECT" "|" "$LICENSE_STRING" "|" "$URL" "|" )
+        LICENSE_STRING=""           # reset LICENSE_STRING
+        IFS=$'\n'
+        continue        # skip to next iteration/line
+    fi
+
+    # closing bracket, lets subtract from OPEN_BRACKET
+    if [[ $line == *"licenses ["* ]]; then
+        OPEN_BRACKET=$(expr 1)
+        continue        # skip to next iteration/line
+    fi
+
+    # Keep appending license names until we hit closing bracket
+    if [[ $OPEN_BRACKET -eq 1 ]]; then
+        LICENSE_STRING+=", $line"
+        continue        # skip to next iteration/line
+    fi
+
+    # This line contains our library, version, and URL
+    if [[ $OPEN_BRACKET -eq 0 ]] && [[ ${#line} -gt 3 ]]; then
+        IFS=$' '
+        read -r PROJECT VERSION URL <<<"$line"
+        PROJECT=$(echo "$PROJECT" | sed 's/"//g')
+        VERSION=$(echo "$VERSION" | sed 's/"//g')
+        URL=$(echo "$URL" | sed 's/"//g')
+        URL="${URL/registry+}"      # remove 'registry+' from start of URL
+        IFS=$'\n'
+    fi
+
+done
```

Comment:
1. JQ has a feature to get all keys. The command below should retrieve all Licence, Crate name and URL
```bash
INPUT=$(jq "." $1)

regex="(.*) [0-9]+\.[0-9]+\.[0-9]+ \((.*)\+(.*)\)"

jq --raw-output "keys[]" <<< $INPUT |
while IFS= read -r key; do
    license=$(jq --arg key "$key" --raw-output '.[$key].licenses | join(", ")' <<< $INPUT)

    if [[ $key =~ $regex ]]; then
        echo "Package name = ${BASH_REMATCH[1]}"
        echo "Pulled from registry or path = ${BASH_REMATCH[2]}"
        echo "URL of crate = ${BASH_REMATCH[3]}"
        echo "License = \"$license\""
    fi
    break
done
```

2. If crate is pulled from path, we should ignore them as they are either examples or Ockam crates

Reply:
Hi, sorry for the late reply. I'm going to finish this up. Do I submit a new PR? Sorry this is my first time submitting a PR.

Reply:
not sure which is easier for you, but any (updating this PR with a new commit, or creating a new PR) works fine.